app/scripts/models/account.js

```diff
@@ -810,7 +810,10 @@ define(function (require, exports, module) {
           .fail(function (err) {
             // If no oauth token existed, or it has gone stale,
             // get a new one and retry.
-            if (ProfileClient.Errors.is(err, 'UNAUTHORIZED')) {
+            if (ProfileClient.Errors.is(err, 'INVALID_TOKEN')) {
+              // invalid token can happen if user uses 'Disconnect' in Firefox Desktop.
+              self.set('accessToken', null);
```

Comment:
When is INVALID_TOKEN returned, only when the sessionToken is invalid?

A couple of notes:

1. Can you use unset instead of set to null?
1. Should we also unset `sessionToken` and `sessionTokenContext`?